homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +81,112 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        return self._var_units
+        return MONITORED_CONDITIONS[self._sensor_type][1]
 
     @property
     def state(self):
         """Return the state of the device."""
-        try:
-            return round(self._api.data[self._var_id], 2)
-        except TypeError:
-            return self._api.data[self._var_id]
+        if self._sensor_type == CURRENT_WARNING_SENSOR:
+            return self._api.current_warning_level
+        return self._api.expected_warning_level
 
     @property
     def device_state_attributes(self):
         """Return the state attributes of the DWD-Weather-Warnings."""
-        data = {ATTR_ATTRIBUTION: ATTRIBUTION, "region_name": self._api.region_name}
-
-        if self._api.region_id is not None:
-            data["region_id"] = self._api.region_id
-
-        if self._api.region_state is not None:
-            data["region_state"] = self._api.region_state
-
-        if self._api.data["time"] is not None:
-            data["last_update"] = dt_util.as_local(
-                dt_util.utc_from_timestamp(self._api.data["time"] / 1000)
-            )
+        data = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_REGION_NAME: self._api.warncell_name,
+            ATTR_REGION_ID: self._api.warncell_id,
+        }
+
+        if self._api.last_update is not None:
+            data[ATTR_LAST_UPDATE] = dt_util.as_local(self._api.last_update)
+        else:
+            data[ATTR_LAST_UPDATE] = None
 
-        if self._var_id == "current_warning_level":
-            prefix = "current"
-        elif self._var_id == "advance_warning_level":
-            prefix = "advance"
+        if self._sensor_type == CURRENT_WARNING_SENSOR:
+            searched_warnings = self._api.current_warnings
         else:
-            raise Exception("Unknown warning type")
-
-        data["warning_count"] = self._api.data[f"{prefix}_warning_count"]
-        i = 0
-        for event in self._api.data[f"{prefix}_warnings"]:
-            i = i + 1
-
-            # dictionary for the attribute containing the complete warning as json
-            event_json = event.copy()
-
-            data[f"warning_{i}_name"] = event["event"]
-            data[f"warning_{i}_level"] = event["level"]
-            data[f"warning_{i}_type"] = event["type"]
-            if event["headline"]:
-                data[f"warning_{i}_headline"] = event["headline"]
-            if event["description"]:
-                data[f"warning_{i}_description"] = event["description"]
-            if event["instruction"]:
-                data[f"warning_{i}_instruction"] = event["instruction"]
-
-            if event["start"] is not None:
+            searched_warnings = self._api.expected_warnings
+
+        data[ATTR_WARNING_COUNT] = len(searched_warnings)
+
+        for i, warning in enumerate(searched_warnings, 1):
```

Comment:
Does this data really change every time `device_state_attributes` is accessed? It's more common to change updating attributes in the update method.

Reply:
Yes you are right, the data doesn't normaly change everytime `device_state_attributes` is accessed, but like at some other places i tried to keep the changes for this PR as small as possible and therefore i didn't change things like that.